embrace-gradle-plugin/src/main/java/io/embrace/android/gradle/plugin/instrumentation/AsmTaskRegistration.kt

```diff
@@ -41,27 +41,29 @@ class AsmTaskRegistration : EmbraceTaskRegistration {
                 params.shouldInstrumentOnLongClick.set(behavior.instrumentation.onLongClickEnabled)
                 params.shouldInstrumentOnClick.set(behavior.instrumentation.onClickEnabled)
 
-                // Find the Asm transformation task by name and make it depend on encodeSharedObjectFilesTask
-                val encodeSharedObjectFilesTask = project.tryGetTaskProvider(
-                    "${EncodeSharedObjectFilesTask.NAME}${data.name.capitalizedString()}",
-                    EncodeSharedObjectFilesTask::class.java
-                ) ?: return@transformClassesWith
+                project.afterEvaluate {
```

Comment:
What testing have we done around this to confirm that it works?

Reply:
Mostly the existing tests and debugging in different setups. Any particular test you think we should add? Tests do break without this 😅

Reply:
I'm happy with that level of testing - the smali decompilation tests should be enough